### PR TITLE
Feature/face generation script

### PR DESCRIPTION
**Pull request by pkrucz00**
What has been done:
 - the basic (pretty much POC) script that opens, clicks over the game, generates faces and their DNAs then shuts the game.
 - a small set of 10 faces and DNAs were generated as an example
 - simple CLI

What should be discussed:
  - should all the people be bold?

What can be done:
  - the faces from different countries have different randomization spectra - the characters generated with "random" button look different than those in Spain, Poland, Sweden or Asia. For now "Poland" is hardcoded. We need a way of mittigating that.
  - choosing different sexes
  - smaller waiting times between positioning th mouse and clicking it

### CODE REVIEW

File: generate_faces.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+@click.option("--faces_folder", type=click.Path(dir_okay=True, file_okay=False),
+                default="faces",
+                help="Output folder for generated faces")
+def main(exec, n, dna_folder, faces_folder):
+    start_game(exec)
+    
+    click_to_the_character_customization_panel()
+    for _ in range(n):
+        model_name = f"{randint(0, 1_000_000)}"
+        click_random_dna()
```
Comment (czyjtu) [inline generate_faces.py]: looking for button location every iteration adds significant overhead. Maybe we can assume that window location will not change during the data collection phase?

File: generate_faces.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    button_placement = pyautogui.locateOnScreen(pic_name, confidence=0.9)
+    while not button_placement:
+        print(f"{pic_name} not found. Retrying in {retry_in_sec} s.")
+        time.sleep(retry_in_sec)
+        button_placement = pyautogui.locateOnScreen(pic_name, confidence=0.9)
+    
+    print(f"Clicking {label}")
+    button_center = pyautogui.center(button_placement)
+    pyautogui.moveTo(button_center.x, button_center.y)
+    time.sleep(1)
```
Comment (czyjtu) [inline generate_faces.py]: why?

File: generate_faces.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    for _ in range(n):
+        model_name = f"{randint(0, 1_000_000)}"
+        click_random_dna()
+        copy_and_save_dna(dna_folder, model_name)
+        screenshot_and_save_face(faces_folder, model_name)
+    
+    end_game()
+
+if __name__=="__main__":
+    main()
```
Comment (czyjtu) [inline generate_faces.py:197]: passing arguments through command-line would be useful (https://docs.python.org/3/library/argparse.html).

File: generate_faces.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                help="Output folder for generated faces")
+@click.option("--faces_folder", type=click.Path(dir_okay=True, file_okay=False),
+                default="faces",
+                help="Output folder for generated faces")
+def main(exec, n, dna_folder, faces_folder):
+    start_game(exec)
+    
+    click_to_the_character_customization_panel()
+    for _ in range(n):
+        model_name = f"{randint(0, 1_000_000)}"
```
Comment (czyjtu) [inline generate_faces.py]: `randint` is fine, but `datetime.now` string looks better IMO

File: generate_faces.py [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    wait_and_click("locate_pics/show_hair_and_beard.png", label="Show Hair and Beard")
+    wait_and_click("locate_pics/head_and_neck.png", label="Head and Neck")
+
+def click_to_the_character_customization_panel():
+    click_through_the_menu()
+    click_through_new_game_start_menu()
+    click_to_the_head_and_neck_panel()
+
+def click_random_dna():
+    wait_and_click("locate_pics/randomize.png", label="Random", wait_in_sec=3)
```
Comment (czyjtu) [inline generate_faces.py]: Why do we have to wait so long?

File: generate_faces.py (hunk 3 above)
Comment (czyjtu) [reply generate_faces.py:197]: done with click

**Review by czyjtu — CHANGES_REQUESTED** [review]

File: generate_faces.py (hunk 4 above)
Comment (pkrucz00) [reply generate_faces.py]: Changed it

File: generate_faces.py (hunk 5 above)
Comment (pkrucz00) [reply generate_faces.py]: The time has been reduced to 0.01 s and the script seems to work properly. I'll try to lower it even further, but I think it is enoug for now.

File: generate_faces.py (hunk 2 above)
Comment (pkrucz00) [reply generate_faces.py]: Using the default function that moved the cursor and immediatly clicked the button was not working properly. GUI sometimes was still focused on the previous button and clicked that instead, which in turn broke the flow of the script. A small wait time (0.05 sec for now but probably less in future) is necessary for it to work properly.

**Comment by pkrucz00** [comment]
What's changed:
   - applying suggested changes (including optimisation)
   - adding more diverce data generation by using choosing multiple sexes and ethnicities 
   - generation time measurement added
   - flag for archiving the results have been added

**Pull request merged**